app/controllers/visitors_details_controller.rb

```diff
@@ -1,22 +1,54 @@
 class VisitorsDetailsController < ApplicationController
   include CookieGuard
   include SessionGuard::OnEditAndUpdate
-  include VisitorsManipulator
+  include TrimParams
+
   before_action :ensure_visit_integrity, only: [:edit, :update]
+  before_action :logstasher_add_visit_id_from_session
 
   def edit
+    visit.visitors = [Visitor.new] if visit.visitors.empty?
     @collect_phone_number = true
   end
 
-  def this_path
-    edit_visitors_details_path
+  def update
+    m = params[:next].match(/remove-(\d)/)
```

Comment:
I can't seem to see anything in the repo that sets the `next` param to the remove + visitor index for this to work which makes me think you always just have `next` being equal to 'Complete' and because the JS has removed the fields from the DOM when the remove link is clicked the code just works by rebuilding the visitors again in the backend from the that requests params. I hope I'm just being stupid here and missing the obvious.